### PR TITLE
Added unit test for pulp login

### PR DESCRIPTION
**Pull request by breillyrh**

### CODE REVIEW

**Comment by breillyrh** [comment]
Just realized this one needs a revision... will break out into its own test "CLI" class.

File: tests/test_pulp_instance.py [hunk 1]
```diff
@@ -268,3 +289,19 @@ def test_checkBlobsFail(self, pulp, repo, blob):
             .and_raise(requests.exceptions.ConnectionError))
         response = pulp.checkBlobs(repo, blobs)
         assert response['error']
+
+# cli tests
+    def test_pulplogin(self):
```
Comment (lcarva) [inline tests/test_pulp_instance.py]: Maybe a new `test_cli.py` file?

Comment (lcarva) [inline tests/test_pulp_instance.py]: Would be great to test the different code paths in `pulp_login` method.

File: tests/test_pulp_instance.py [hunk 2]
```diff
@@ -268,3 +289,19 @@ def test_checkBlobsFail(self, pulp, repo, blob):
...
+        bopts = testbOpts()
+        p = testPulp()
+        flexmock(Pulp)
+        (Pulp
+            .should_receive('__new__')
+            .once()
+            .and_return(p))
+        flexmock(os.path)
+        (os.path
```
Comment (lcarva) [inline tests/test_pulp_instance.py]: There's a way to use flexmock to specify the args that should have been used for each of the calls.

File: tests/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# tests
+class TestCLI(object):
+    # Tests of methods from CLI
+    @pytest.mark.parametrize('server, config_file, debug, cert, key, error',
+                             [('testserv', 'testconf', None, None, None, False),
+                              ('testserv', 'testconf', True, None, None, False),
+                              ('testserv', 'testconf', None, True, True, False),
+                              ('testserv', 'testconf', None, None, None, True)])
+    def test_pulplogin(self, server, config_file, debug, cert, key, error):
```
Comment (lcarva) [inline tests/test_cli.py]: Should probably be `test_pulp_login`.

File: tests/test_cli.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        return
+
+    def setDebug():
+        return
+
+
+# tests
+class TestCLI(object):
+    # Tests of methods from CLI
+    @pytest.mark.parametrize('server, config_file, debug, cert, key, error',
```
Comment (lcarva) [inline tests/test_cli.py]: Since `server` and `config_file` never change, I suggest just making local variables in the `test_pulp_login` method.

Also, `debug` seems unrelated to `cert`, `key`. And `error` seems unrelated as well. You could have multiple `parametrize` decorators:

```python
    @pytest.mark.parametrize('debug', [True, False])
    @pytest.mark.parametrize('error', [True, False])
    @pytest.mark.parametrize('cert, key', [...])
    def test_pulplogin(self, debug, cert, key, error):
```

File: tests/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+# tests
+class TestCLI(object):
+    # Tests of methods from CLI
+    @pytest.mark.parametrize('debug', [True, False])
+    @pytest.mark.parametrize('error', [True, False])
+    @pytest.mark.parametrize('cert, key', [(None, True), (True, None), (True, True), (None, None)])
+    def test_pulp_login(self, debug, cert, key, error):
+        bopts = testbOpts("testserv", "testconf", debug, cert, key)
+        p = testPulp()
+        flexmock(Pulp).new_instances(p) \
```
Comment (twaugh) [inline tests/test_cli.py]: Recommended way of splitting this over multiple lines is like this:
```
(flexmock(Pulp)
    .new_instances(p)
    .with_args(...))
```

File: tests/test_cli.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            .with_args(Pulp, env=bopts.server, config_file=bopts.config_file)
+        if debug:
+            flexmock(testPulp)
+            (testPulp
+                .should_receive('setDebug')
+                .once()
+                .and_return(None))
+        flexmock(os.path)
+        if error and not (key or cert):
+            (os.path
```
Comment (twaugh) [inline tests/test_cli.py]: This looks odd. Shouldn't this be the `Expectation` that flexmock(os.path) returns?
